lib/market.rb

```diff
@@ -0,0 +1,43 @@
+class Market
+  attr_reader :name, :vendors
+
+  def initialize(name)
+    @name = name
+    @vendors = []
+  end
+
+  def add_vendor(vendor)
+    @vendors << vendor
+  end
+ 
+  def vendor_names
+    @vendors.map do |vendor|
+      vendor.name
+    #binding.pry
+    end
+  end
+  def vendors_that_sell(item)
+    @vendors.find_all do |vendor|
+      vendor.inventory.key?(item)
+    end
+  end
+
+  def total_inventory
```

Comment:
Remember what this method is looking for. We are creating a hash where the values are the keys are the items and another hash as the value - this sub hash is going to tell us the total quantity of that item and its vendors. Think about going through each item and using the code youve already written to get the vendors and the quantities to build this sub hash.